src/VtkGeometryRenderer.js

```diff
@@ -0,0 +1,112 @@
+/* global window */
+
+import React                from 'react';
+
+// import sizeHelper           from 'paraviewweb/src/Common/Misc/SizeHelper';
+
+import vtkOpenGLRenderWindow           from 'vtk.js/Sources/Rendering/OpenGL/RenderWindow';
+import vtkSynchronizableRenderWindow   from 'vtk.js/Sources/Rendering/Misc/SynchronizableRenderWindow';
+import vtkRenderWindowInteractor       from 'vtk.js/Sources/Rendering/Core/RenderWindowInteractor';
+
+const SYNCHRONIZATION_CONTEXT_NAME = 'pvwLocalRenderingContext';
+
+export default React.createClass({
+
+  displayName: 'VtkGeometryRenderer',
+
+  propTypes: {
+    className: React.PropTypes.string,
+    client: React.PropTypes.object,
+    viewId: React.PropTypes.number,
+    interactionTimout: React.PropTypes.number,
+    connection: React.PropTypes.object,
+    showFPS: React.PropTypes.bool,
+    style: React.PropTypes.object,
+  },
+
+  getDefaultProps() {
+    return {
+      className: '',
+      showFPS: false,
+      style: {},
+      viewId: -1,
+      interactionTimout: 500,
+    };
+  },
+
+  componentWillMount() {
+    // Make sure we monitor window size if it is not already the case
+    console.log('Mounting VtkGeometryRenderer');
+    // sizeHelper.startListening();
+  },
+
+  componentDidMount() {
+    console.log('VtkGeometryRenderer is now mounted');
+
+    // Get our hands on the default synchronization context and tell how to fetch data arrays
+    const synchCtx = vtkSynchronizableRenderWindow.getSynchronizerContext(SYNCHRONIZATION_CONTEXT_NAME);
```

Comment:
The context name should be a prop with a default so you don't need to provide it.